v3/subgraphs/goerli/src/market.ts

```
@@ -0,0 +1,83 @@
+import {
+  MarketRegistered,
+  MarketManagerModule,
+  MarketUsdDeposited,
+  MarketUsdWithdrawn,
+} from '../generated/MarketManagerModule/MarketManagerModule';
+import { Market } from '../generated/schema';
+import { BigDecimal, log } from '@graphprotocol/graph-ts';
+import { createMarketSnapshotByBlock } from './marketSnapshotByBlock';
+import { createMarketSnapshotByDay } from './marketSnapshotByDay';
+import { createMarketSnapshotByWeek } from './marketSnapshotByWeek';
+
+export function handleMarketCreated(event: MarketRegistered): void {
+  const newMarket = new Market(event.params.marketId.toString());
+  newMarket.address = event.params.market;
+  newMarket.created_at = event.block.timestamp;
+  newMarket.created_at_block = event.block.number;
+  newMarket.updated_at = event.block.timestamp;
+  newMarket.updated_at_block = event.block.number;
+  newMarket.usd_deposited = BigDecimal.fromString('0');
+  newMarket.usd_withdrawn = BigDecimal.fromString('0');
+  newMarket.net_issuance = BigDecimal.fromString('0');
+  newMarket.reported_debt = BigDecimal.fromString('0');
+  newMarket.save();
+}
+
+export function handleMarketUsdDeposited(event: MarketUsdDeposited): void {
+  const marketId = event.params.marketId.toString();
+  const market = Market.load(marketId);
+  if (market == null) {
+    log.error(
+      'Something went wrong, got a MarketUsdDeposited event for a market that doesnt exists ' +
+        event.params.marketId.toString(),
+      []
+    );
+    return;
```

weird that you don't get a build error since the function is `void`. Maybe wrap the reset into an else? But up to your gusto

---

`return` nothing is `void`

---

I prefer an early return for this edge case, we could even throw since this would be an unexpected error. But an error log is quite visible in the subgraphs ui

---

ah okay cool, nvm then